pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/invertedindex/RealtimeLuceneIndexReaderRefreshThread.java

```diff
@@ -0,0 +1,105 @@
+ * It encapsulate a lock and all the realtime lucene readers for the particular realtime segment.
+ * Since text index is also create on a per column basis, there will be as many realtime lucene
+ * readers as the number of columns with text search enabled.
+ *
+ * Between each successive execution of the task, there is a fixed delay (regardless of how long
+ * each execution took). When the task wakes up, it pick the RealtimeLuceneReadersForRealtimeSegment
+ * from the head of queue, refresh it's readers and adds this at the tail of queue.
+ */
+public class RealtimeLuceneIndexReaderRefreshThread implements Runnable {
+  private static final Logger LOGGER = LoggerFactory.getLogger(SegmentColumnarIndexCreator.class);
+  public static final int INITIAL_DELAY_MS_DEFAULT = 1000;
+  // TODO: make this configurable and choose a higher default value
+  public static final int DELAY_BETWEEN_SUCCESSIVE_EXECUTION_MS_DEFAULT = 10;
+
+  private final ConcurrentLinkedQueue<RealtimeLuceneReadersForRealtimeSegment> _luceneRealtimeReaders;
+
+  public RealtimeLuceneIndexReaderRefreshThread(ConcurrentLinkedQueue<RealtimeLuceneReadersForRealtimeSegment> luceneRealtimeReaders) {
+    _luceneRealtimeReaders = luceneRealtimeReaders;
+  }
+
+  @Override
+  public void run() {
+    RealtimeLuceneReadersForRealtimeSegment realtimeReadersForSegment = _luceneRealtimeReaders.poll();
+    if (realtimeReadersForSegment != null) {
+      String segmentName = realtimeReadersForSegment.getSegmentName();
+      if (!realtimeReadersForSegment.isSegmentAboutToBeDestroyed()) {
+        // If segmentAboutToBeDestroyed is false, it implies MutableSegmentImpl
+        // code hasn't yet started to destroy the segment. So the refresh task
+        // can proceed for now.
+        boolean refreshed = true;
+        if (realtimeReadersForSegment.getLock().tryLock()) {
+          // got the lock
+          try {
+            List<RealtimeLuceneTextIndexReader> realtimeLuceneReaders = realtimeReadersForSegment.getRealtimeLuceneReaders();
+            for (RealtimeLuceneTextIndexReader realtimeReader : realtimeLuceneReaders) {
+              if (realtimeReadersForSegment.isSegmentAboutToBeDestroyed()) {
+                // Check this here to know if MutableSegmentImpl wants to start destroying the
+                // realtime segment but is unable to proceed since we hold the lock. So we will stop
+                // in the middle, abort the refresh and let MutableSegmentImpl proceed with destroying
+                // the segment.
+                refreshed = false;
+                break;
+              }
+              SearcherManager searcherManager = realtimeReader.getSearcherManager();
+              try {
+                searcherManager.maybeRefresh();
+              } catch (Exception e) {
+                LOGGER.warn("Caught exception {} while refreshing realtime lucene reader for segment: {}", e, segmentName);
```

Comment:
One too many `{}`. Also, e should be the last argument

Reply:
please fix this. what is the likelihood of this exception?